narwhal/node/tests/reconfigure.rs

```diff
@@ -97,37 +96,42 @@ impl ExecutionState for SimpleExecutionState {
 
 impl SimpleExecutionState {
     async fn process_transaction(&self, transaction: Transaction, change_epoch: bool) {
-        let _transaction: u64 = bincode::deserialize(&transaction).unwrap();
+        let transaction: u64 = bincode::deserialize(&transaction).unwrap();
         // Change epoch every few certificates. Note that empty certificates are not provided to
         // this function (they are immediately skipped).
         let mut epoch = self.committee.lock().unwrap().epoch();
-        if change_epoch {
+        if transaction >= epoch && change_epoch {
             epoch += 1;
-            {
-                let mut guard = self.committee.lock().unwrap();
-                guard.epoch = epoch;
-            };
-
-            let worker_keypairs = self.worker_keypairs.iter().map(|kp| kp.copy());
-            let worker_ids = 0..self.worker_keypairs.len() as u32;
-            let worker_ids_and_keypairs = worker_ids.zip(worker_keypairs).collect();
-
-            let new_committee = self.committee.lock().unwrap().clone();
-
-            self.tx_reconfigure
-                .send((
-                    self.keypair.copy(),
-                    self.network_keypair.copy(),
-                    new_committee,
-                    worker_ids_and_keypairs,
-                    self.worker_cache.clone(),
-                ))
-                .await
-                .unwrap();
+
+            self.send_new_epoch_reconfigure(epoch).await;
         }
 
         let _ = self.tx_output.send(epoch).await;
     }
+
+    async fn send_new_epoch_reconfigure(&self, epoch: Epoch) {
+        {
+            let mut guard = self.committee.lock().unwrap();
+            guard.epoch = epoch;
+        }
+
+        let worker_keypairs = self.worker_keypairs.iter().map(|kp| kp.copy());
+        let worker_ids = 0..self.worker_keypairs.len() as u32;
+        let worker_ids_and_keypairs = worker_ids.zip(worker_keypairs).collect();
```

Comment:
```suggestion
        let worker_ids_and_keypairs = worker_keypair.iter().cloned().enumerate().collect();
```

Reply:
Ah great reduction - I didn't pay too much attention to this